thrift/src/main/java/com/linecorp/armeria/common/thrift/text/StructContext.java

```diff
@@ -132,35 +138,85 @@ protected TField getTFieldByName(String name) throws TException {
      * TProtocol.writeStructBegin(), rather than relying on the stack trace.
      */
     private static Class<?> getCurrentThriftMessageClass() {
-        final StackTraceElement[] frames =
-                Thread.currentThread().getStackTrace();
-
-        for (StackTraceElement f : frames) {
-            final String className = f.getClassName();
-
+        if (SystemInfo.javaVersion() >= 9) {
+            Optional<Class> result = Optional.empty();
             try {
-                final Class<?> clazz = Class.forName(className);
-
-                // Note, we need to check
-                // if the class is abstract, because abstract class does not have metaDataMap
-                // if the class has no-arg constructor, because FieldMetaData.getStructMetaDataMap
-                //   calls clazz.newInstance
-                if (isTBase(clazz) && !isAbstract(clazz) && hasNoArgConstructor(clazz)) {
-                    return clazz;
-                }
+                final Class<?> stackWalker = Class.forName("java.lang.StackWalker");
+                final Object stackWalkerInstance = stackWalker
+                        .getDeclaredMethod("getInstance")
+                        .invoke(null);
+                final Method walk = stackWalkerInstance.getClass().getDeclaredMethod("walk", Function.class);
```

Comment:
Could we use `MethodHandle` which is way faster than Java reflection? https://www.baeldung.com/java-method-handles